sssom/util.py

```diff
@@ -273,6 +273,8 @@ def filter_redundant_rows(
     :param ignore_predicate: If true, the predicate_id column is ignored, defaults to False
     :return: Filtered pandas DataFrame
     """
+    if CONFIDENCE not in df.columns:
```

Comment:
What would happen if we assumed that missing confidence means 100% confidence? I think this assumption is more often true. Also, if you do ad the IF clause here, the dataframe will not be sorted?

Reply:
I just realized, line 280 has `assign_default_confidence` which adds a new column 'confidence' if absent BUT it initiates to `np.NaN`. Is this what we still intend to do?

Reply:
I think so, but it is important that the absence of a confidence value does not lead to a removal of the entire mapping! While it is not the same as confidence 1.0, it should probably be interpreted as such during reconciliation.. Unless you see a strong reason against it?

Reply:
The presence of `np.NaN` is causing the df returned to be empty at the codes [between lines 297 to 311](https://github.com/mapping-commons/sssom-py/blob/issue-348/sssom/util.py#L297-L311)

Reply:
Would it make sense to itiniate it to "0.0" instead of `np.Nan`